torch/distributed/distributed_c10d.py

```diff
@@ -805,7 +855,11 @@ def init_process_group(
     elif init_method is None:
         init_method = "env://"
 
-    backend = Backend(backend)
+    if backend:
+        backend = Backend(backend)
+    else:
+        backend = Backend("undefined")
+    backend_config = BackendConfig(backend)
```

Comment:
For the purpose of multi-str parsing, we should directly pass the input arg `backend` to `backend_config`.